app/ts/components/pages/WebsiteAccess.tsx

```diff
@@ -0,0 +1,561 @@
+		</div>
+	)
+}
+
+const RemoveAddressConfirmation = ({ websiteOrigin, address }: { address: bigint, websiteOrigin: string }) => {
+	const { searchQuery } = useWebsiteAccess()
+	const addressString = serialize(EthereumAddress, address)
+
+	const removeAddressAccessForWebsite = async () => {
+		await updateWebsiteAccess((existingAccessList) => {
+			const newWebsiteAccessMap = new Map(existingAccessList.map(access => [access.website.websiteOrigin, { ...access }]))
+			if (newWebsiteAccessMap.has(websiteOrigin)) {
+				const websiteAccessOfWebsiteOrigin = newWebsiteAccessMap.get(websiteOrigin)!
+				if (!websiteAccessOfWebsiteOrigin.addressAccess) return existingAccessList
+				const addressAccessOfWebsiteOriginMap = new Map(websiteAccessOfWebsiteOrigin.addressAccess.map(addressAccess => [addressAccess.address, { ...addressAccess }]))
+				if (addressAccessOfWebsiteOriginMap.has(address)) addressAccessOfWebsiteOriginMap.delete(address)
+				websiteAccessOfWebsiteOrigin.addressAccess = Array.from(addressAccessOfWebsiteOriginMap.values())
+			}
+			return Array.from(newWebsiteAccessMap.values())
+		})
+
+		sendPopupMessageToBackgroundPage({ method: 'popup_retrieveWebsiteAccess',  data: { query: searchQuery.value } })
+	}
+
+	const confirmOrRejectRemoval = (returnValue: string) => {
+		if (returnValue !== 'confirm') return
+		removeAddressAccessForWebsite()
+	}
+
+	return (
+		<Modal>
+			<Modal.Open class = 'btn btn--ghost'><TrashIcon /></Modal.Open>
+			<Modal.Dialog class = 'dialog' style = { { textAlign: 'center', color: 'var(--disabled-text-color)' } } onClose = { confirmOrRejectRemoval }>
+				<h2 style = { { fontWeight: 600, fontSize: '1.125rem', color: 'var(--text-color)', marginBlock: '1rem' } }>Removing Address</h2>
+				<p style = { { marginBlock: '0.5rem' } }>This will prevent <pre>{websiteOrigin}</pre> from accessing to the following address <data value = { addressString } style = { { backgroundColor: 'var(--card-bg-color)', color: 'var(--text-color)', padding: '2px 4px', borderRadius: 2 } }>{ addressString }</data></p>
+				<p style = { { marginBlock: '1rem' } }>Are you sure you want to continue?</p>
+				<div style = { { display: 'flex', flexWrap: 'wrap', columnGap: '1rem', justifyContent: 'center', marginBlock: '1rem' } }>
+					<Modal.Close class = 'btn btn--outline' value = 'reject'>Cancel</Modal.Close>
+					<Modal.Close class = 'btn btn--destructive' value = 'confirm'>Confirm</Modal.Close>
+				</div>
+
+			</Modal.Dialog>
+		</Modal>
+	)
+}
+
+const AddressCard = ({ address }: { address: bigint }) => {
+	return (
+		<article class = 'flexy flexy-sm'>
+			<figure><Blockie style = { { fontSize: '2rem' } } address = { address } /></figure>
+			<section style = { { flex: 1 } }>
+				<p style = { { color: 'var(--text-color)' } }>vitalik.eth</p>
+				<BigIntToEthereumAddress bigIntAddress = { address } class = 'truncate' style = { { fontSize: '0.875rem', color: 'var(--disabled-text-color)' } } />
```

Comment:
we already have small address component that also can be edited, this cannot be edited

Reply:
created a separate PR #1118, to help pull edit functionality to addresses